### ISSUE TITLE
Tugboat Disk Size Critical

### ISSUE THREAD

**Issue by 7hunderbird** [department-of-veterans-affairs/va.gov-cms#18343]
## User Story or Problem Statement

Yesterday we got an alert in Slack that the Tugboat disk size was getting critical.  Upon reviewing the datadog monitor I saw that we were at the warning state of 75% utilized and wanted to create this ticket so that we can reach out to Tugboat support (`#tugboat-support`) and understand if there are commands that we can run to truncate or trim the file usage.

- Slack / PagerDuty alert

  ![CleanShot 2024-06-18 at 10 50 59](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/3761/e6361289-7e5c-464b-b0bf-020731a1b787)

- Datadog monitor shows 75% utilization.

  ![CleanShot 2024-06-18 at 10 56 16](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/3761/b6c4e91a-f03f-4c45-a1bb-1ae1c087710f)

- [PagerDuty alert](https://dsva.pagerduty.com/incidents/Q1NADRAC04KY8Q?utm_campaign=channel&utm_source=slack)
- [Datadog monitor](https://vagov.ddog-gov.com/monitors/183289?view=spans)

## Description or Additional Context

I'd like to pursue this issue with either @jcbolling or @nfpappas-oddball or both.  With the goal not only of resolution but sharing the knowledge of how we interface with Tugboat and how to do supoprt for it.

## Steps for Implementation
- [x] determine who will reach out to tugboat support in DSVA slack
- [x] share this ticket and details with tugboat support
- [x] fix the disk usage on their recommenation
- [x] thank them for a job well done and give them a positive emoji reaction in slack 🎉 

## Acceptance Criteria
- [x] Disk usage is below warning of 75% utilization
- [x] We understand steps or proceedures of how to reduce disk usage
- [x] We've documented them in confluence

**Comment by 7hunderbird** [comment]
The tugboat EC2 instance has 3 volumes attached.  The third is an 8 TB volume and it's 75% full.

**Comment by gracekretschmer-metrostar** [comment]
Next steps: @7hunderbird will reach out to the tugboat support channel.

**Comment by gracekretschmer-metrostar** [comment]
Tyler: the fix that was working in the lower environment is not working in upper environments, so actively troubleshooting.

**Comment by 7hunderbird** [comment]
Thread in slack:

- https://dsva.slack.com/archives/C01A35JDH88/p1719592167424139

Did a `docker info` to see usage

```
Data Space Used: 6.146TB
Data Space Total: 8.774TB
Data Space Available: 2.628TB
```

> can you run the following command which will delete any containers that are no longer linked to Tugboat.

```
for x in $(docker ps -a | awk '{print $NF}' | grep -v ^tugboat- | grep -v NAMES); do tugboat --no-color ls $x -q; done 2>&1 | grep not\ found | awk '{print $1}' | xargs docker rm -f
```

(we have a cleanup script in `/opt/tugboat/util/clean-agent.sh` that will do the commands we’re going to work through, but I’d rather take this step by step manually to see where the discrepancy is… and given the size of the discrepancy, that script might take a LONG time to complete as well)

`/opt/tugboat/util/clean-agent.sh`

  

Once you’re in an empty directory somewhere, run:

```
docker images | awk '{print $2}' | sed 's/_anchor//g' | sed 's/_root//g' | sed 's/_container//g' | sort | uniq | grep -E "[0-9a-f]{24}" > ids.txt
```

Then:

```
docker images | awk '{print $1}' | grep -E "[0-9a-f]{24}" >> ids.txt
```

Then:

```
for x in $(cat ids.txt | sort | uniq); do tugboat --no-color ls $x -q; done 2>&1 | grep not\ found | awk '{print $1}' > not-found.txt
```

Then check if there’s anything in `not-found.txt` with cat or less or whatever

To remove those, can you run:

```
for x in `cat not-found.txt`; do docker images | grep $x | awk '{print $1 ":" $2 }'; done | xargs docker rmi
```

okay, next can you run `docker image prune`

**Comment by 7hunderbird** [comment]
Updated https://vfs.atlassian.net/wiki/spaces/PCMS/pages/3192160305/Tugboat+Trouble+Shooting with instructions to run `/opt/tugboat/util/clean-agent.sh` command
